test/1_stdlib/Runtime.swift

```diff
@@ -729,108 +365,23 @@ Runtime.test("_stdlib_atomicCompareExchangeStrongPtr") {
-  expectEqual("_TtGC1a12GenericClassVS_11PlainStruct_",
-              NSStringFromClass(GenericClass<PlainStruct>.self))
-  expectEqual("_TtGC1a12GenericClassOS_9PlainEnum_",
-              NSStringFromClass(GenericClass<PlainEnum>.self))
-  expectEqual("_TtGC1a12GenericClassTVS_11PlainStructOS_9PlainEnumS1___",
-              NSStringFromClass(GenericClass<(PlainStruct, PlainEnum, PlainStruct)>.self))
-  expectEqual("_TtGC1a12GenericClassMVS_11PlainStruct_",
-              NSStringFromClass(GenericClass<PlainStruct.Type>.self))
-  expectEqual("_TtGC1a12GenericClassFMVS_11PlainStructS1__",
-              NSStringFromClass(GenericClass<PlainStruct.Type -> PlainStruct>.self))
-
-  expectEqual("_TtGC1a12GenericClassFzMVS_11PlainStructS1__",
-              NSStringFromClass(GenericClass<PlainStruct.Type throws -> PlainStruct>.self))
-  expectEqual("_TtGC1a12GenericClassFTVS_11PlainStructROS_9PlainEnum_Si_",
-              NSStringFromClass(GenericClass<(PlainStruct, inout PlainEnum) -> Int>.self))
-
-  expectEqual("_TtGC1a12GenericClassPS_9ProtocolA__",
-              NSStringFromClass(GenericClass<ProtocolA>.self))
-  expectEqual("_TtGC1a12GenericClassPS_9ProtocolAS_9ProtocolB__",
-              NSStringFromClass(GenericClass<protocol<ProtocolA, ProtocolB>>.self))
-  expectEqual("_TtGC1a12GenericClassPMPS_9ProtocolAS_9ProtocolB__",
-              NSStringFromClass(GenericClass<protocol<ProtocolA, ProtocolB>.Type>.self))
-  expectEqual("_TtGC1a12GenericClassMPS_9ProtocolAS_9ProtocolB__",
-              NSStringFromClass(GenericClass<protocol<ProtocolB, ProtocolA>.Protocol>.self))
-
-  expectEqual("_TtGC1a12GenericClassCSo7CFArray_",
-              NSStringFromClass(GenericClass<CFArray>.self))
-  expectEqual("_TtGC1a12GenericClassVSC9NSDecimal_",
-              NSStringFromClass(GenericClass<NSDecimal>.self))
-  expectEqual("_TtGC1a12GenericClassCSo8NSObject_",
-              NSStringFromClass(GenericClass<NSObject>.self))
-  expectEqual("_TtGC1a12GenericClassCSo8NSObject_",
-              NSStringFromClass(GenericClass<NSObject>.self))
-  expectEqual("_TtGC1a12GenericClassPSo9NSCopying__",
-              NSStringFromClass(GenericClass<NSCopying>.self))
-  expectEqual("_TtGC1a12GenericClassPSo9NSCopyingS_9ProtocolAS_9ProtocolB__",
-              NSStringFromClass(GenericClass<protocol<ProtocolB, NSCopying, ProtocolA>>.self))
-
-  expectEqual("_TtGC1a17MultiGenericClassGVS_13GenericStructSi_GOS_11GenericEnumGS2_Si___",
-              NSStringFromClass(MultiGenericClass<GenericStruct<Int>,
-                                                  GenericEnum<GenericEnum<Int>>>.self))
-}
-
 Runtime.test("casting AnyObject to class metatypes") {
   do {
     var ao: AnyObject = SomeClass()
     expectTrue(ao as? Any.Type == nil)
     expectTrue(ao as? AnyClass == nil)
-
-    ao = SomeNSObjectSubclass()
-    expectTrue(ao as? Any.Type == nil)
-    expectTrue(ao as? AnyClass == nil)
-
-    ao = SomeClass.self
```

Comment:
It doesn't work on Linux because line 799 is ill-typed. Conversions from metatypes to AnyObject are only possible when objective-c interop is enabled. See CSSimplify.cpp [lines 1673-1681](https://github.com/apple/swift/blob/aa5105c4bf87fd77b2b2124285196e158777218c/lib/Sema/CSSimplify.cpp#L1673-L1681) which is guarded on [line 1632](https://github.com/apple/swift/blob/aa5105c4bf87fd77b2b2124285196e158777218c/lib/Sema/CSSimplify.cpp#L1632).